File: cpp/2319.check-if-matrix-is-x-matrix.cpp
#include <vector>

using std::vector;

class Solution {
public:
  bool checkXMatrix(vector<vector<int>> &grid) {
    int m = grid.size(), n = grid.back().size();

    if (m != n) {
      return false;
    }

    for (int i = 0; i < m; i++) {
      for (int j = 0; j < n; j++) {
        if ((i == j) || (m - i - 1 == j)) {
          if (grid[i][j] == 0) {
            return false;
          }
        } else {
          if (grid[i][j] != 0) {
            return false;
          }
        }
      }
    }

    return true;
  }
};

File: golang/2319.check-if-matrix-is-x-matrix.go
func checkXMatrix(grid [][]int) bool {
	m, n := len(grid), len(grid[0])
	if m != n {
		return false
	}

	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if (i == j) || (n-i-1 == j) {
				if grid[i][j] == 0 {
					return false
				}
			} else {
				if grid[i][j] != 0 {
					return false
				}
			}
		}
	}

	return true
}
